implementation/leprechaun/src/shared/api/http-exceptions.ts

```diff
@@ -0,0 +1,69 @@
+import { HttpException, HttpStatus } from '@nestjs/common';
+import { ApiProperty } from '@nestjs/swagger';
+
+enum ErrorKind {
+  SchemaMismatchException = 'schema-mismatch-exception',
+  NotFoundException = 'resource-not-found-exception',
+  InternalServerError = 'internal-server-error',
+}
+
+interface AppError {
+  kind: ErrorKind;
+  code: number;
+  messageEn: string;
+  messagePl?: string;
+}
+
+export class HttpExceptionDto implements AppError {
+  @ApiProperty({ description: 'Error kind', enum: ErrorKind, isArray: false })
+  kind: ErrorKind;
+  @ApiProperty({ description: 'HTTP status code' })
+  code: number;
+  @ApiProperty({ description: 'Error message in English' })
+  messageEn: string;
+  @ApiProperty({ description: 'Error message in Polish', required: false })
+  messagePl?: string;
+
+  constructor(kind: ErrorKind, code: number, messageEn: string, messagePl: string) {
+    this.kind = kind;
+    this.code = code;
+    this.messageEn = messageEn;
+    this.messagePl = messagePl;
+  }
+}
```

Comment:
**nit:** The current approach lets you create errors with mismatched `kind` and `code`, which can be prevented with TS's type system.

Reply:
I'll turn this enum into object the same way I did for Healthcheck, and I'll keep `(kind, status_code)` as a tuple for creating DTOs.